hdat/archive.py

```diff
@@ -50,20 +56,30 @@ def select_recent(self, i, *args):
         else:
             return self.read_result(os.path.join(top_directory, recent_id))
 
-    def select_recents_suite(self, *args):
+    def select_recents_suite(self, suites, *args):
+        """Return the most recent result of each case within a specified suite.
+        *args -- suite ID selector ['a']"""
         top_directory = os.path.join(self.root, *args)
         if not os.path.isdir(top_directory):
             msg = "Selected suite directory {} does not exist or is not a directory"
             raise AbortError(msg.format(top_directory))
-
+        # catch unused cases within a suite when resultspec is an entire suite
+        for case in suites[str(args[0])].collect().keys():
+            if case not in os.listdir(top_directory):
+                msg = 'The case "{}" exists within suite "{}", ' + \
+                      'but has no result recorded. ' + \
+                      'Please run the case or suite first.'
+                raise AbortError(msg.format(case, args[0]))
         for entry in os.listdir(top_directory):
-            if not entry.startswith('.') and os.path.isdir(os.path.join(top_directory, entry)):
-                yield self.select_recent(-1, *(args+(entry,)))
+            if (not entry.startswith('.') and
```

Comment:
I would split this line into a few intermediate variables in order to avoid so much wrapping.